workspaces/liferay-ticket-workspace/client-extensions/liferay-ticket-custom-element/src/services/tickets.ts

```diff
@@ -123,3 +122,50 @@ export async function generateNewTicket() {
 		method: 'POST',
 	});
 }
+
+export async function updateTicketStatus(ticket: Ticket) {
+	ticket.payload.ticketStatus =
+		listTypeDefinitions[J3Y7_STATUSES].map[ticket.ticketStatus];
+
+	const result = await fetch(`/o/c/j3y7tickets/${ticket.id}`, {
+		body: JSON.stringify(ticket.payload),
+		headers: {
+			'accept': 'application/json',
+			'content-Type': 'application/json',
+			'x-csrf-token': Liferay.authToken,
+		},
+		method: 'PUT',
+	});
+
+	if (result.ok) {
+		return;
+	}
+	else {
+		const jsonResult = await result.json();
+
+		throw new Error(`${JSON.stringify(jsonResult)}`);
+	}
+}
+
+export async function assignTicketToMe(ticket: Ticket) {
+	const result = await fetch(
+		`/o/c/j3y7tickets/by-external-reference-code/${ticket.externalReferenceCode}/object-actions/AssignTicketToMe`,
+		{
+			headers: {
```

Comment:
There's a lot of duplicate code for your requests... a helper function could simplify this. Ask Mahmoud to show you how I helped him with this.